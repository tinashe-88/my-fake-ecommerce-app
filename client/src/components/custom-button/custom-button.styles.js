import styled, { css } from 'styled-components'

const transitionAnimation = css`
  transition: background-color .15s ease-in;
  transition: color .15s ease-in;
`

const buttonStyles = css`
  background-color: #000;
  color: white;
  border: none;

  &:hover {
    background-color: #fff;
    color: #000;
    border: 1px solid #000;
    transition: ${transitionAnimation};
  }
`

const invertedButtonStyles = css`
  background-color: #fff;
  color: #000;
  border: 1px solid #000;

  &:hover {
    background-color: #000;
    color: #fff;
    transition: ${transitionAnimation};
  }
`

const googleSignInStyles = css`
  background-color: #4285f4;
  color: #fff;

  &:hover {
    background-color: #358ae8;
    border: none;
    transition: ${transitionAnimation};
  }
`

const getButtonStyles = props => {
  if (props.isGoogleSignIn){
    return googleSignInStyles
  }

  return props.inverted ? invertedButtonStyles : buttonStyles;
}

export const CustomButtonContainer = styled.button`
  min-width: 165px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 25px;
  font-size: 15px;
  background-color: #000;
  color: #fff;
  text-transform: uppercase;
  font-family: 'Lato', sans-serif;
  font-weight: bolder;
  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;

  @media(max-width: 800px){
    margin: 10px 0;
  }

  ${getButtonStyles}
`