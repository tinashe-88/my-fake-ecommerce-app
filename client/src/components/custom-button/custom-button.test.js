import React from 'react'
import { shallow } from 'enzyme'

import { CustomButton } from './custom-button'

it('Should render CustomButtom component', () => {
  expect(shallow(<CustomButton />)).toMatchSnapshot()
})