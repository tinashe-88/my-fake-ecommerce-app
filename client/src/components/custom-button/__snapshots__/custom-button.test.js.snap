// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CustomButtom component 1`] = `<styled.button />`;
